e2e-tests/README.md

```diff
@@ -0,0 +1,114 @@
+
+Run the following commands to install the required components:
+
+```
+sudo yum -y install epel-release https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+sudo yum -y install coreutils sed jq curl docker
+sudo curl -s -L https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -o /usr/bin/yq
+sudo chmod a+x /usr/bin/yq
+curl -s -L https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz \
+    | tar -C /usr/bin --strip-components 1 --wildcards -zxvpf - '*/oc' '*/kubectl'
+curl -s https://get.helm.sh/helm-v3.2.4-linux-amd64.tar.gz \
+    | tar -C /usr/bin --strip-components 1 -zxvpf - '*/helm'
+curl https://sdk.cloud.google.com | bash
+```
+
+### MacOS
+
+Install [Docker](https://docs.docker.com/docker-for-mac/install/), and run the following commands for the other required components:
+
+```
+brew install coreutils gnu-sed jq kubernetes-cli openshift-cli kubernetes-helm
+brew install yq@3
+brew link yq@3
+curl https://sdk.cloud.google.com | bash
+```
+
+### Runtime requirements
+
+Also, you need a Kubernetes platform of [supported version](https://docs.percona.com/percona-operator-for-postgresql/System-Requirements.html#officially-supported-platforms), available via [GKE](https://docs.percona.com/percona-operator-for-postgresql/gke.html), [OpenShift](https://docs.percona.com/percona-operator-for-postgresql/openshift.html) or [minikube](https://docs.percona.com/percona-operator-for-postgresql/minikube.html) to run the Operator.
+
+**Note:** there is no need to build an image if you are going to test some already-released version.
+
+## Building and testing the Operator
+
+There are scripts which build the image and run tests. Both building and testing
+needs some repository for the newly created docker images. If nothing is
+specified, scripts use Percona's experimental repository `perconalab/percona-postgresql-operator`, which
+requires decent access rights to make a push.
+
+To specify your own repository for the Operator docker image, you can use IMAGE environment variable:
+
+```
+export IMAGE=bob/my_repository_for_test_images:K8SPG-42-fix-feature-X
+```
+We use linux/amd64 platform by default. To specify another platform, you can use DOCKER_DEFAULT_PLATFORM environment variable
+
+```
+export DOCKER_DEFAULT_PLATFORM=linux/amd64
+```
+
+Use the following script to build the image:
+
+```
+./e2e-tests/build
+```
+
+You can also build the image and run your cluster in one command:
+
+```
+./e2e-tests/build-and-run
+```
+
+Running all tests at once can be done with the following command:
+
+```
+./e2e-tests/run
+```
+
+(see how to configure the testing infrastructure [here](#using-environment-variables-to-customize-the-testing-process)).
+
+Tests can also be run one-by-one using the appropriate scripts (their names should be self-explanatory):
+
+```
+./e2e-tests/affinity/run
+./e2e-tests/clone-cluster/run
+./e2e-tests/data-migration-gcs/run
```

Comment:
data-migration is an obsolete test. We are keeping it just for history. Here is an actual test list

`affinity
clone-cluster
demand-backup
init-deploy
monitoring
ns-mode
operator-self-healing
recreate
scaling
scheduled-backup
self-healing
smart-update
tls-check
upgrade
users
version-service`